src/services/checkpoints/ShadowCheckpointService.ts

```diff
@@ -92,12 +91,18 @@ export abstract class ShadowCheckpointService extends EventEmitter {
 			throw new Error("Shadow git repo already initialized")
 		}
 
-		const hasNestedGitRepos = await this.hasNestedGitRepositories()
+		const nestedGitPath = await this.getNestedGitRepository()
+
+		if (nestedGitPath) {
+			// Show persistent error message with the offending path
+			const relativePath = path.relative(this.workspaceDir, nestedGitPath)
+			const message = t("common:errors.nested_git_repos_warning", { path: relativePath })
+			vscode.window.showErrorMessage(message)
 
-		if (hasNestedGitRepos) {
 			showWarning(t("kilocode:checkpoints.nestedGitRepos")) // kilocode_change
```

Comment:
This is now a double warning I guess? I think ours might be better because it tries to not be shown too often.